src/zenml/steps/base_step.py

```diff
@@ -333,6 +341,14 @@ def _prepare_input_artifacts(
                     f"already passed as a positional argument."
                 )
 
+            if not isinstance(artifact, Channel):
+                raise StepInterfaceError(
+                    f"Wrong argument type (`{type(artifact)}`) for argument "
```

Comment:
Can't this also be like a `BaseStepConfig`?

Reply:
I think in `BaseStep.__call__`  we are inside the pipeline connect function, in which case only Channels/Artifacts are allowed. Configs would be passed to `BaseStep.__init__` instead when creating the step.

Reply:
Correct! My bad. Resolving